src/hooks/use-list-behavior.js

```diff
@@ -0,0 +1,72 @@
+import { useState, useCallback } from 'react'
+
+/**
+ * Hook that returns bind props + accessibility props for
+ * "selectable" elements
+ */
+
+const resetStyles = {
+  outline: 'none'
+}
+
+export function useListBehavior({
+  disabled,
+  isHighlighted,
+  isSelectable,
+  isSelected,
+  onSelect,
+  ref: inputRef
+}) {
+  const [ref, setRef] = useState(inputRef)
```

Comment:
Using state for refs will trigger unnecessary rerenders. Would recommend using `useRef`